src/Streamly/Internal/Data/Fold.hs

```diff
@@ -563,6 +563,11 @@ defaultSalt = 0x087fc72c
 rollingHash :: (Monad m, Enum a) => Fold m a Int
 rollingHash = rollingHashWithSalt defaultSalt
 
+
+{-# INLINABLE rollingHashFirstN #-}
+rollingHashFirstN :: (Monad m, Enum a) => Int -> Fold m a Int
```

Comment:
It will be nice to have brief haddock documentation about these combinators. It will also be nice to have some simple tests for these.